### ISSUE TITLE
Improve flutter contrast test coverage

### ISSUE THREAD

**Issue by ankur2136** [flutter/flutter#109465]
## Use case
Looking at the contrast check here (packages/flutter_test/lib/src/accessibility.dart), the text elements which are repeated more than once throw a failure. In order to get away from this failure we need to bypass contrast test on all cases where we have repeated elements.  

Product GP3, Money

## Proposal

Loop over all elements in the semantic node and check for contrast of each element. 
This might have some repeated loops but should terminate.
